limitador-server/src/main.rs

```diff
@@ -219,6 +249,40 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
     };
 
+    let _watcher = if let Ok(limits_file_path) = env::var(LIMITS_FILE_ENV) {
```

Comment:
nothing to be done with the watcher? maybe register a signal to stop watching?

Reply:
Well that's what this does… RAII… i.e. we keep it around, and things get cleaned up upon `drop()`, e.g. in the case of linux, that'd be [here](https://github.com/notify-rs/notify/blob/main/src/inotify.rs#L644-L650)

Reply:
btw, this is very much like automatic `defer`al as you'd know it from golang… only you _cannot_ forget to tell it to release the resources!

Reply:
Ok, let me verify if I understood. 

When `main()` function run, the execution point is blocked at 

```rust
run_http_server(&http_api_address, rate_limiter.clone()).await?;
```

When the process gets killed (with a signal), the execution continues in `main` returning `Ok(())`. Then, `_watcher` which is an Option holding the watcher, gets out of scope and dropped. Notify lib will clean up at that moment. 

is that right?

Reply:
Yes that's exactly it. Or to be real pedantic, while you hit the return, the function doesn't return until `.drop()` has itself returned on the `Option` (that if it owns the `Watcher` eventually delegates to its `drop()`)… 

The [doc](https://doc.rust-lang.org/std/ops/trait.Drop.html) might provide more clarity than my explanation here… otherwise (and I recommend it in all cases) [Rust for Rustaceans](https://nostarch.com/rust-rustaceans) is excellent, and the [free first chapter](https://nostarch.com/download/samples/RustforRustaceans_Ch1.pdf) has the part on `Drop` and drop order on page 9!